### DISCUSSION TITLE
Will it be possible to get the text for my current rule in validator?

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1680] [Q&A] [answered]
Hi Langium community, I am wondering will it be possible to get the text for my current rule in validator?

e.g. The text before operator and after operator in assignment.

Or I need to loop through the astnode to find the complete text before operator.

**Comment by msujew** [comment, marked as answer]
Hey @durianwaffle,

You can use the `node.$cstNode?.text` property to access the original text of any AST node.

**Reply by durianwaffle** [reply]
Thank you @msujew !
